test/Datadog.Trace.Tests/Util/CleanUriSegmentTests.cs

```diff
@@ -0,0 +1,22 @@
+using Xunit;
+
+namespace Datadog.Trace.Tests.Util
+{
+    public class CleanUriSegmentTests
+    {
+        [Theory]
+        [InlineData("/", "/")]
+        [InlineData("/controller/action/b37855d4bae34bd3b3357fc554ad334e", "/controller/action/?")]
+        [InlineData("/controller/action/14bb2eed-34f0-4aa2-b2c3-09c0e2166d4d", "/controller/action/?")]
+        [InlineData("/controller/action/14bb2eed-34f0X4aa2-b2c3-09c0e2166d4d", "/controller/action/14bb2eed-34f0X4aa2-b2c3-09c0e2166d4d")]
+        [InlineData("/controller/action/14bb2eed-34f0-4aa2Xb2c3-09c0e2166d4d", "/controller/action/14bb2eed-34f0-4aa2Xb2c3-09c0e2166d4d")]
+        [InlineData("/controller/action/14bb2eed-34f0A4aa2Bb2c3C09c0e2166d4d", "/controller/action/14bb2eed-34f0A4aa2Bb2c3C09c0e2166d4d")]
+        [InlineData("/DataDog/dd-trace-dotnet/blob/e2d83dec7d6862d4181937776ddaf72819e291ce/src/Datadog.Trace/Util/UriHelpers.cs", "/DataDog/dd-trace-dotnet/blob/e2d83dec7d6862d4181937776ddaf72819e291ce/src/Datadog.Trace/Util/UriHelpers.cs")]
```

Comment:
This is out of scope of this PR, but this test data makes me think we should replace these other cases as well, even though they don't specifically parse as GUIDs or integers. Maybe we should expand our logic to any hexadecimal number as well (which any number/position of dashes).

For example, the commit hash in
```
/DataDog/dd-trace-dotnet/blob/e2d83dec7d6862d4181937776ddaf72819e291ce/src/Datadog.Trace/Util/UriHelpers.cs
```
should be removed to reduce resource name cardinality:
```
/DataDog/dd-trace-dotnet/blob/?/src/Datadog.Trace/Util/UriHelpers.cs
```

Reply:
Another thought is to add a user setting where they can specify a regex to match custom urls. For example something like `/country/spain/edit` may be `/country/?/edit`, but we would never catch that without user input or a route template.

Anyway, just something to think about.

Reply:
I agree with both ideas. A little concern in performance impact supporting regex values but yeah, I think is useful a feature like that.